## Statistical Analysis of Starbucks Nutritional Drink Data
Julia Beitel - Big Data - September 14, 2018



### Abstract 
In this lab I will be analyizing the Starbucks Nutritional Drink data. Within my analysis, I will be answering the following questions that the data posed: Which drink has the most and least of each nutrional section? How many different drinks are there? What percentage of drinks have 2% milk?

### Dataset Preparation
The given data represents the following columns: Beverage category, beverage name, beverage preparation, calories, total fat, trans fat, saturated fat, sodium, total carbs, cholestorol, dietary fiber, sugar, protein, vitamin A %, vitamin C %, iron %, and caffeine.
This data was uploaded on July 30th of 2018 from the following source: https://www.kaggle.com/starbucks/starbucks-menu

#### Open and Download

In [26]:
#first i opened the file
data_file = open("starbucks_drinkMenu_expanded.csv", "r");

#then i created an empty list to store the nutritional data
drink_data = [];

#then i put all columns from the drink data file into our list
for ii in data_file:
    drink_data.append(ii);
    
#then i closed the file
data_file.close();
print(drink_data[:5])

['Beverage_category,Beverage,Beverage_prep,Calories, Total Fat (g),Trans Fat (g) ,Saturated Fat (g), Sodium (mg), Total Carbohydrates (g) ,Cholesterol (mg), Dietary Fibre (g), Sugars (g), Protein (g) ,Vitamin A (% DV) ,Vitamin C (% DV), Calcium (% DV) ,Iron (% DV) ,Caffeine (mg)\n', 'Coffee,Brewed Coffee,Short,3,0.1,0,0,0,5,0,0,0,0.3,0%,0%,0%,0%,175\n', 'Coffee,Brewed Coffee,Tall,4,0.1,0,0,0,10,0,0,0,0.5,0%,0%,0%,0%,260\n', 'Coffee,Brewed Coffee,Grande,5,0.1,0,0,0,10,0,0,0,1,0%,0%,0%,0%,330\n', 'Coffee,Brewed Coffee,Venti,5,0.1,0,0,0,10,0,0,0,1,0%,0%,2%,0%,410\n']


#### Removing Columns and Sorting to Lists

Next, I removed the column header and the other unwanted columns of data. These columns include vitamin A %, vitamin C %, and iron %. Then, I  split the data into the following  lists: Beverage category, beverage name, beverage preparation, calories, total fat, trans fat, saturated fat, sodium, total carbs, cholestorol, dietary fiber, sugar, protein, and caffeine.
I did not clean up any commas, nor did I edit the original csv file in any way. Additionally, I did not delete any columns, I merely did not creat lists for the following columns: trans fat, saturated fat, dietary fiber, vitamin A %, vitamin C %, and iron %.

In [27]:
#here i removed the first element (this is the column header)
del drink_data[0];

#then i createed empty lists for each of the different corresponding columns
beverage_category = [];
drink_name = [];
beverage_prep = [];
calories = [];
total_fat = [];
sodium = [];
cholesterol = [];
sugar = [];
protein = [];
caffeine = [];

for jj in range(len(drink_data)):
    #next i (using a for loop and the .split fxn) split each of the data units using commas to separate them
    drink_data[jj] = drink_data[jj].split(",")
    
    #then i added the data to each specific list for it's corresponding column
    beverage_category.append(str(drink_data[jj][0]))
    drink_name.append(str(drink_data[jj][1]))
    beverage_prep.append(str(drink_data[jj][2]))
    calories.append(float(drink_data[jj][3]))
    total_fat.append(drink_data[jj][4])
    sodium.append(drink_data[jj][7])
    cholesterol.append(float(drink_data[jj][9]))
    sugar.append(float(drink_data[jj][11]))
    protein.append(float(drink_data[jj][12]))
    caffeine.append(drink_data[jj][17])

print(beverage_category[:10])

['Coffee', 'Coffee', 'Coffee', 'Coffee', 'Classic Espresso Drinks', 'Classic Espresso Drinks', 'Classic Espresso Drinks', 'Classic Espresso Drinks', 'Classic Espresso Drinks', 'Classic Espresso Drinks']


### Data Modeling 
First, I found the drink with the highest and lowest numbers of nutritional data in each section...

#### Min Nutritional Data

In [28]:
#here i used the min() fxn to find the min overall 
min_cal = min(calories)
cal_index = calories.index(min_cal)
print("Min calories:", drink_name[cal_index])

min_fat = min(total_fat)
fat_index = total_fat.index(min_fat)
print("Min fat:", drink_name[fat_index])

min_sodium = min(sodium)
sodium_index = sodium.index(min_sodium)
print("Min sodium:", drink_name[sodium_index])

min_chol = min(cholesterol)
chol_index = cholesterol.index(min_chol)
print("Min cholesterol:", drink_name[chol_index])

min_sugar = min(sugar)
sugar_index = sugar.index(min_sugar)
print("Min sugar:", drink_name[sugar_index])

min_protein = min(protein)
protein_index = protein.index(min_protein)
print("Min protein:", drink_name[protein_index])

min_caff = min(caffeine)
caffeine_index = caffeine.index(min_caff)
print("Min caffeine:", drink_name[caffeine_index])

Min calories: Tazo® Tea
Min fat: Caffè Americano
Min sodium: Brewed Coffee
Min cholesterol: Brewed Coffee
Min sugar: Brewed Coffee
Min protein: Caramel Apple Spice (Without Whipped Cream)
Min caffeine: Iced Brewed Coffee (With Milk & Classic Syrup)


#### Max Nutritional Data

In [29]:
#here i used the max() fxn to find the max overall 
max_cal = max(calories)
max_cal_index = calories.index(max_cal)
print("Max calories:", drink_name[max_cal_index])

max_fat = max(total_fat)
max_fat_index = total_fat.index(max_fat)
print("Max fat:", drink_name[max_fat_index])

max_sodium = max(sodium)
max_sodium_index = sodium.index(max_sodium)
print("Max sodium:", drink_name[max_sodium_index])

max_chol = max(cholesterol)
max_chol_index = cholesterol.index(max_chol)
print("Max cholesterol:", drink_name[max_chol_index])

max_sugar = max(sugar)
max_sugar_index = sugar.index(max_sugar)
print("Max sugar:", drink_name[max_sugar_index])

max_protein = max(protein)
max_protein_index = protein.index(max_protein)
print("Max protein:", drink_name[max_protein_index])

max_caff = max(caffeine)
max_caffeine_index = caffeine.index(max_caff)
print("Max caffeine:", drink_name[max_caffeine_index])

Max calories: White Chocolate Mocha (Without Whipped Cream)
Max fat: Caffè Latte
Max sodium: Caffè Latte
Max cholesterol: Java Chip (Without Whipped Cream)
Max sugar: Java Chip (Without Whipped Cream)
Max protein: Banana Chocolate Smoothie
Max caffeine: Tazo® Full-Leaf Tea Latte


#### Number of Total Drinks

In [30]:
print("There are", len(drink_name), "Starbucks drinks.")

There are 242 Starbucks drinks.


#### Percent of 2% Milk Drinks

In [31]:
count = 0
for x in beverage_prep:
    if x == "2% Milk":
        count = count + 1
print("There are", count, "drinks that use 2% Milk.")
total = count/len(drink_name)
print("This makes up %.2f%% of drinks."% (total*100))

There are 50 drinks that use 2% Milk.
This makes up 20.66% of drinks.


### Data Analysis 

Will my modeling I found that the drink which consistently showed up within the minimum nutritional data analysis was the Brewed Coffee. The Brewed Coffee had the least sodium, cholesterol, and sugar of all 242 total starbucks drinks (I also found this total number).
I also found that drink that had the the most within the maximum nutritional data analysis was the Java Chip Frappachino and the Caffè Latte. The Caffè Latte had the most fat and sodium, while the Java Chip Frappachino had the most cholesterol and sugar. 
This implies that the least healthy drinks are the Caffè Latte and the Java Chip Frappachino, while the healthiest drink is the Brewed Coffee.
This conclusion makes logical, valid sense because the Brewed Coffee has the least added ingredients and the concluded "unhealthy" drinks have a lot of ingredients. 

I also found that the percentage of drinks that included 2% milk is 20.66. This goes to show that if you are lactose intolerant, you have around a 4/5 chance of getting a drink that will not upset your allergy. Luckily, you always have the convienent opportunity to ask for a dairy substitute. 

### Conclusion

This data raises many questions. Some of them being: Is it hard to dine at Starbucks and be vegan? What does the food nutritional data show? What is the most unhealthy meal possible at Starbucks? 

The data also posed questions towards the validity and legitamacy of the data collection, as it did in the last lab. Some being: Are there any flaws in the data? How accurate are these numbers? Has the nutritional data changed since this data was collected? 
It would be interesting to dive deeper into these questions because, living in Seattle, there seems to be a Starbucks on every corner. 

### Acknowledgements
I also used sample code for my data preparation and analysis from Ms.Sconyers's Python intro document on jupyter notebooks. I recieved help from Ms.Sconyers with simplifying long integers through the %.2f system. Lillian Ellis helped me to sort my list indexes through the .index() function. Again, I recieved my Starbucks Nutritional Data from the following source: https://www.kaggle.com/starbucks/starbucks-menu